pyannote/audio/interactive/diarization/recipe.py

```diff
@@ -0,0 +1,430 @@
+# The MIT License (MIT)
+#
+# Copyright (c) 2021-2022 CNRS
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+# AUTHORS
+# Jim Petiot
+# Hervé Bredin
+
+import base64
+import functools
+import random
+from collections.abc import Iterator
+from copy import deepcopy
+from datetime import datetime
+from functools import lru_cache
+from itertools import groupby
+from pathlib import Path
+from tempfile import mkstemp
+from typing import Any, Dict, Iterable, List, Union
+
+import numpy as np
+import prodigy
+import torch
+from frozendict import frozendict
```

Comment:
This dependency is missing in `requirements.txt`.

That being said, it would be nice to __not__ rely on yet another dependency.
Can you remove the caching mechanism for now?